R/plotting.R

```diff
@@ -1,225 +1,90 @@
-      .data$value, .data$forecaster, .data$forecast_date,
-      .data$target_end_date
+# Version: 2024-04-23
+
+plot_forecasts <- function(predictions_cards, forecast_date, exclude_geos, geo_type, data_7dav_override = NULL) {
+  assert_that(nrow(predictions_cards) > 0)
+  assert_that(geo_type %in% c("state", "nation"))
+
+  start_day <- as.Date("2022-10-01")
+
+  if (is.null(data_7dav_override)) {
+    signal_data <- epidatr::pub_covidcast(
+      "hhs",
+      "confirmed_admissions_influenza_1d_7dav",
+      geo_type,
+      "day",
+      "*",
+      epirange(start_day, forecast_date)
     )
-
-  lower_bounds <- predictions_cards %>%
-    select(.data$quantile) %>%
-    filter(.data$quantile < 0.5) %>%
-    unique() %>%
-    pull()
-  quantiles_to_plot <- as.integer(sort(
-    round(500L * (1 + intervals %o% c(-1L, 1L)))
-  ))
-
-  quantiles_df <- predictions_cards %>%
-    filter(as.integer(round(.data$quantile * 1000)) %in% c(quantiles_to_plot)) %>%
-    mutate(
-      endpoint_type = if_else(.data$quantile < 0.5, "lower", "upper"),
-      alp = if_else(.data$endpoint_type == "lower",
-        format(2 * .data$quantile, digits = 3, nsmall = 3),
-        format(2 * (1 - .data$quantile), digits = 3, nsmall = 3)
-      ),
-      interval = forcats::fct_rev(
-        paste0((1 - as.numeric(.data$alp)) * 100, "%")
-      )
-    ) %>%
-    select(-.data$quantile, -.data$alp) %>%
-    pivot_wider(names_from = "endpoint_type", values_from = "value")
-
-  return(quantiles_df)
-}
-
-get_points_df <- function(predictions_cards) {
-  points_df <- predictions_cards %>%
-    filter(as.integer(round(.data$quantile * 1000)) == 500L |
-      is.na(.data$quantile))
-  if (any(is.na(points_df$quantile))) {
-    points_df <- points_df %>%
-      pivot_wider(names_from = "quantile", values_from = "value") %>%
-      mutate(value = if_else(!is.na(.data$`NA`), .data$`NA`, .data$`0.5`)) %>%
-      select(-.data$`0.5`, -.data$`NA`)
   } else {
-    points_df <- points_df %>%
-      select(-.data$quantile)
+    signal_data <- data_7dav_override %>%
+      filter(geo_value != "us", start_day <= time_value)
   }
+  signal_data <- signal_data %>%
+    filter(!.data$geo_value %in% exclude_geos) %>%
+    select(.data$geo_value, .data$time_value, .data$value) %>%
+    mutate(
+      data_source = "hhs",
+      forecaster = "hhs hosp truth",
+      value = 7L * .data$value
+    ) %>%
+    rename(target_end_date = .data$time_value)
 
-  return(points_df)
-}
-
-plot_quantiles <- function(g, quantiles_df) {
-  n_quantiles <- nlevels(quantiles_df$interval)
-  l_quantiles <- levels(quantiles_df$interval)
-
-  alp <- c(.4, .2, .1)
-  for (qq in n_quantiles:1) {
+  # Setup plot
+  g <- ggplot(signal_data, mapping = aes(
+    x = .data$target_end_date,
+    color = .data$forecaster,
+    fill = .data$forecaster
+  )) +
+    geom_line(mapping = aes(y = .data$value))
+
+  # Plot (symmetric) quantiles
+  quantiles <- c(0.8, 0.95)
+  alphas <- c(0.4, 0.2)
```

Comment:
```suggestion
  alphas <- c(0.4, 0.1)
```
??

Reply:
Previous code, after some convoluted calculations, plotted the quantiles: .05, .25, .5, .75, .95. The .5 quantile was drawn with a thick line, the .75 had an alpha (opacity) value of .4 and .95 was .2. The .1 was unused and besides the .1 is very faint and hard to see. I'll update the 0.8 quantile here to match the previous value of .75 though.